### DISCUSSION TITLE
Farmer mahcine become plot machine and secondary machine becomes principal Farmer : how to do

### DISCUSSION THREAD

**Discussion by casa10** [Chia-Network/chia-blockchain#3407] [Support]
Hello

I have curently one machine who do all (Farmer, plotter and wallet). it is the laptop on the schema who is linked to a NAS for example.
I would like to make another machine (deskstop computer) only farmer and to use my first machine (the laptop) as plotter.

Laptop become plotter
Desktop become principal farmer and can make plots too

How i can do please ? i don't know with the different keys.
I know to plot only public farmer key and pulbic pool key are necessary.
But to be farmer what is necessary and how to do.
I put schema from how i am now and what i want. If someone can help me. 

In the schema, laptop is alone in the top and it's ma first machine who do all.
At the end, i want this laptop become only plotter and my secondary machine becoe farmer only or farmer and plotter.

![image](https://user-images.githubusercontent.com/8081413/116811725-0f265780-ab4b-11eb-9cb0-f3c56d5c1165.png)

**Comment by yapwenbin** [comment]
Just setup your new Desktop machine as normal, then `chia keys add` and input your 20 mnemonic phrase.

You can stop all the services on your Laptop, then run it as a dedicated plotter (or disable upnp if they are on the same network to prevent issues)

**Reply by casa10** [reply]
Thank you.
I will try like that so :)
- stop all services on laptop
- install chia on deskstop machine
-  import from mnemonics to the deskstop machine
-  chia keys add (the public farmer key and pulbic pool key) on deskstop machine
- run the deskstop machine
- run laptop and disable upnp on the laptop.

Don't need to copy config/ssl/ca from the laptop to the deskstop machine ?
Does The import from mnemnonics creates same certiciatcates on config/ssl/ca on the deskstop machine ?

**Reply by yapwenbin** [reply]
I'm running a few full node + wallet + farmer + harvester with the same mnemonic phrase (All from different network), and I didn't manually copy any certs. All of then runs fine.

**Reply by casa10** [reply]
Me all my computer will be in same network.
So. i don't know

**Reply by yapwenbin** [reply]
If not mistaken the mnemonic phrase will restore your keys. Just disable upnp on your laptop then should be no problem running on the same network.

**Reply by casa10** [reply]
ok. i will try. Because i don't know how work this mnemonics.
I will do with mnemonics and after, I will try to check all keys present in config/ssl/directory1, config/ssl/directory2, etc... of the deskstop computer and compare them with the others present in my laptop.
I will disable upnp of the laptop and run.

In config/ssl directory there are directories : ca, farmer, full_node,harvester, timelord, wallet, daemon, introducer.
In each of theses directories, you have different certificates.

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
